spec/models/transformation_mapping_spec.rb

```diff
@@ -117,7 +119,10 @@
       end
 
       it 'returns valid vms' do
-        result = mapping.search_vms_and_validate(['name' => vm.name])
+        this_vm = FactoryBot.create(:vm_vmware, :name => 'this_test_vm', :ems_cluster => src, :ext_management_system => FactoryBot.create(:ext_management_system))
```

Comment:
Could this be defined in a `let(:vm)` at the context level instead of in each test?

Reply:
No.   "this_vm" is specific to each test.

Reply:
But isn't it exactly the same in every test?  Also you're creating a new ems everytime but reusing the same cluster.  The ems should also be a let

Reply:
And if you need something different it can be modified in that context

Reply:
Your are right about `this_vm` being exactly the same.  I tried putting it in its `let(:this_vm)` at the parent context level but then both tests fail.  Behind the scene executions of `let` statements is not clear and how database is modified.  Putting it in each test appears to have better control and tests can pass.

Reply:
`let()` is lazily evaluated, `let!()` is eagerly evaluated so if you need it to be defined but don't actually call it directly from a test you want to use `let!()`

Reply:
Thanks @agrare.  It was `let!()` is what was needed to make it work at the context level.

Reply:
Also, please remove `name` as it should be sequenced by the factory.

Reply:
sorry i responded to this comment before committing my code.